src/App.tsx

```diff
@@ -1,222 +1,137 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import classnames from 'classnames';
+import { createTodos, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filters';
+import { Header } from './components/Auth/Header';
+import { TodoList } from './components/Auth/TodoList';
+import { FooterTodo } from './components/Auth/FooterTodo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [value, setValue] = useState('');
+  const [islinkAll, setislinkAll] = useState(true);
+  const [islinkActive, setislinkActive] = useState(false);
+  const [islinkCompleted, setislinkCompleted] = useState(false);
+
+  const [currentFilter, setCurrentFilter] = useState(Filters.All);
+  const [isError, setisError] = useState(false);
+  const [isHidden, setisHidden] = useState(false);
+  const [completedTodo, setCompletedTodo] = useState<Todo>({
+    id: 0,
+    userId: 0,
+    title: '',
+    completed: false,
+  });
+
+  const handleCompletedTodo = () => {
+    setCompletedTodo((prevState) => ({
+      ...prevState,
+      completed: !prevState.completed,
+    }));
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    setisHidden(true);
+
+    getTodos(user?.id)
+      .then(setTodos)
+      .catch(() => setisError(true));
+
+    setTimeout(() => {
+      setisHidden(true);
+    }, 3000);
+
+    createTodos(value).then(setTodos);
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
+  const filterList = (): Todo[] | undefined => todos.filter((todo) => {
+    switch (currentFilter) {
+      case Filters.Active:
+        setislinkAll(false);
+        setislinkActive(true);
+        setislinkCompleted(false);
```

Comment:
Please, fix my previous comment. You have a current filter, so you don't need 3 additional states for each filter. Let's imagine you have 20 filters or even more, what would you do in this case?